
syntax = "proto3";

message TaskStructure{
  enum Priority {
    NONE = 0;
    FIRST = 1;
    SECOND = 2;
    THIRD = 3;
  }
  string name = 1;
  string label = 2;
  int32 date = 3;
  Priority prior = 4;
  bool completed = 5;

}

message TaskEntity{
  TaskStructure task = 1;
  repeated TaskEntity subtasks = 2;
}

message Storage {
  repeated TaskEntity tasks = 1;
}